posts/banff-cohomology.md

```diff
@@ -0,0 +1,37 @@
+
+Emily Witt gave a talk on local cohomology, a theory which was not formalised in Lean when she gave the talk, but was formalised in Lean later that day by Emily herself and Scott Morrison. The work was merged into mathlib by the end of the week. Emily, Jake Levinson, and Sam van Gool spent some of the week developing a basic API for local cohomology.
+
+María Inés de Frutos Fernández talked about her work on formalising Fontaine theory, some of it with Antoine Chambert-Loir. A consequence of her work is that we will soon be able to state comparison isomorphisms between different cohomology theories which show up in arithmetic. One obstruction to doing this now is the unfortunate fact that we still have no definition of etale cohomology and my impression was that little progress was made here, despite many of the ingredients being available. One area where there was some progress was the theory of sheaves of modules, which is still not in mathlib; I am beginning to wonder whether we will end up with three different implementations of the concept, plus isomorphisms between them, with different choices of definition being useful for different applications.
+
+On the airporter back to Calgary, Heather and I initiated a "Riemann Roch Race". People talk about "the Riemann-Roch theorem" but this is a misnomer -- there are two Riemann-Roch theorems, one a theorem about algebraic line bundles on complete algebraic curves, the other a theorem about complex line bundles on compact Riemann surfaces. Some people might claim that these are "the same thing", the way mathematicians do, but the bottom line is that these are not the same thing unless you prove some theorems like GAGA which are much harder than the Riemann-Roch theorem. To be honest I am surprised that places like Wikipedia do not stress this fact. Claims like "the Grothendieck-Riemann-Roch theorem is a generalisation of the Riemann-Roch theorem" and "The Atiyah-Singer index theorem is a generalisation of the Riemann-Roch theorem" are slightly misleading to the extent that they are generalisations of different theorems; by this point, neither of these theorems imply the other one: they are theorems belonging to two different branches of mathematics. So, given that there are two Riemann-Roch theorems, which one will be formalised first in Lean? Hopefully the algebraists and analysts will take up the challenge, because it is quite clear that we need both. Note that there are several different ways of stating these theorems, some of which use cohomology. Even showing that various different statements are equivalent to each other will be an interesting exercise.
```

Comment:
```suggestion
On the airporter back to Calgary, Heather and I initiated a "Riemann-Roch Race". People talk about "the Riemann-Roch theorem" but this is a misnomer -- there are two Riemann-Roch theorems, one a theorem about algebraic line bundles on complete algebraic curves, the other a theorem about complex line bundles on compact Riemann surfaces. Some people might claim that these are "the same thing", the way mathematicians do, but the bottom line is that these are not the same thing unless you prove some theorems like GAGA which are much harder than the Riemann-Roch theorem. To be honest I am surprised that places like Wikipedia do not stress this fact. Claims like "the Grothendieck-Riemann-Roch theorem is a generalisation of the Riemann-Roch theorem" and "The Atiyah-Singer index theorem is a generalisation of the Riemann-Roch theorem" are slightly misleading to the extent that they are generalisations of different theorems; by this point, neither of these theorems imply the other one: they are theorems belonging to two different branches of mathematics. So, given that there are two Riemann-Roch theorems, which one will be formalised first in Lean? Hopefully the algebraists and analysts will take up the challenge, because it is quite clear that we need both. Note that there are several different ways of stating these theorems, some of which use cohomology. Even showing that various different statements are equivalent to each other will be an interesting exercise.
```